app/javascript/packages/document-capture/components/address-search.tsx

```diff
@@ -0,0 +1,51 @@
+import { TextInput, Button } from '@18f/identity-components';
+import { request } from '@18f/identity-request';
+import { useState, useCallback, ChangeEvent } from 'react';
+
+interface Location {
+  street_address: string;
+  city: string;
+  state: string;
+  zip_code: string;
+  address: string;
+}
+
+interface AddressSearchProps {
+  onAddressFound?: (location: Location) => void;
+}
+
+export const ADDRESS_SEARCH_URL = '/api/addresses';
+
+function AddressSearch({ onAddressFound = () => {} }: AddressSearchProps) {
+  const [unvalidatedAddressInput, setUnvalidatedAddressInput] = useState('');
+  const [addressQuery, setAddressQuery] = useState({} as Location);
+  const handleAddressSearch = useCallback(async () => {
+    const addressCandidates = await request(ADDRESS_SEARCH_URL, {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      json: { address: unvalidatedAddressInput },
+    });
+
+    const [bestMatchedAddress] = addressCandidates;
+    setAddressQuery(bestMatchedAddress);
+    onAddressFound(bestMatchedAddress);
+  }, [unvalidatedAddressInput]);
+
+  return (
+    <>
+      <TextInput
+        value={unvalidatedAddressInput}
+        onChange={(event: ChangeEvent) => {
+          const target = event.target as HTMLInputElement;
+
+          setUnvalidatedAddressInput(target.value);
+        }}
+        label="Search for an address"
+      />
+      <Button onClick={() => handleAddressSearch()}>Search</Button>
```

Comment:
To make the most of `useCallback` memoization, we should avoid creating a new function reference each time here, since otherwise there's no point to memoizing.

```suggestion
      <Button onClick={handleAddressSearch}>Search</Button>
```

Reply:
Just want to note that this is going to be very prone to race conditions if the user submits multiple times, as the results may arrive out of order form when they were sent.

We should either cancel/nullify previous requests like we're doing in the other component, or consider a library like `swc` / `react-query` to handle this on our behalf.

Reply:
The suggestion was applied, but just want to resurface [the race condition concern from my other comment in this thread](https://github.com/18F/identity-idp/pull/7385#discussion_r1036340328). I wouldn't necessarily consider it blocking, but I definitely expect it to produce some inconsistent buggy behavior.